### PR TITLE
created registration page

### PR DESCRIPTION
**Pull request by Nimishaer**

### CODE REVIEW

File: client/src/pages/instructor/Registration.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+export default function RegistrationInstructor() {
+
+  function registration(data) {
+
+  }
+
+  const schema = yup.object().shape({
+    name: yup.string().matches(/^[A-Za-z ]*$/, 'Please enter valid name').required("Name field cannot be empty*"),
+    email: yup.string().required("Email is required*").matches(/^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/g, 'Email must be valid'),
```
Comment (mishelshaji) [inline client/src/pages/instructor/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/instructor/Registration.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    mobile: yup.string()
+    .required("Mobile is required*")
+    .min(10, 'Mobile number must be 10 digits')
+    .max(10, 'Mobile number must be 10 digits')
+    .matches(/^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/, 'Enter valid mobile number'),
+    yearOfExperience: yup.string().required("Field Required*"),
```
Comment (mishelshaji) [inline client/src/pages/instructor/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/instructor/Registration.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    education: yup.string().required("Field required*"),
```
Comment (mishelshaji) [inline client/src/pages/instructor/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/instructor/Registration.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    dob: yup.string("Enter valid date of birth").required("Field required*"),
```
Comment (mishelshaji) [inline client/src/pages/instructor/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/instructor/Registration.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    areaOfExpertise: yup.string().required("Field required*"),
```
Comment (mishelshaji) [inline client/src/pages/instructor/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/instructor/Registration.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    password: yup.string().required("Password cannot be empty*").matches(/^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/, 'Minimum eight characters, at least one letter and one number'),
```
Comment (mishelshaji) [inline client/src/pages/instructor/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/instructor/Registration.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    confirmPassword: yup.string().oneOf([yup.ref('password')], "Passwords doesn't match").required("Confirm password cannot be empty*")
```
Comment (mishelshaji) [inline client/src/pages/instructor/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/student/Registration.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+export default function Registration() {
+
+  function registration(data) {
+
+  }
+
+  const schema = yup.object().shape({
+    name: yup.string().matches(/^[A-Za-z ]*$/, 'Please enter valid name').required("Name field cannot be empty*"),
+    email: yup.string().required("Email is required*").matches(/^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/g, 'Email must be valid'),
```
Comment (mishelshaji) [inline client/src/pages/student/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/student/Registration.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+import { yupResolver } from '@hookform/resolvers/yup';
+
+export default function Registration() {
+
+  function registration(data) {
+
+  }
+
+  const schema = yup.object().shape({
+    name: yup.string().matches(/^[A-Za-z ]*$/, 'Please enter valid name').required("Name field cannot be empty*"),
```
Comment (mishelshaji) [inline client/src/pages/student/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/student/Registration.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    email: yup.string().required("Email is required*").matches(/^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/g, 'Email must be valid'),
+    mobile: yup.string().required("Mobile is required*").min(10,'Mobile number must be 10 digits').max(10,'Mobile number must be 10 digits').matches(/^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/,'Enter valid mobile number'),
```
Comment (mishelshaji) [inline client/src/pages/student/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/student/Registration.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    password: yup.string().required("Password cannot be empty*").matches(/^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,'Minimum eight characters, at least one letter and one number'),
```
Comment (mishelshaji) [inline client/src/pages/student/Registration.js]: Use single quotes to represent string values.

File: client/src/pages/student/Registration.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+              <div className="mb-3">
+                <h2 className="text-center text-secondary mt-5">Register</h2>
+              </div>
+              <div className="mb-3">
+                <label htmlFor="">Name</label>
+                <input type="text" className="form-control" {...register("name")} />
+                {errors.name ? <span className="text-danger">{errors.name.message}</span> : null}
+              </div>
+              <div className="mb-3">
+                <label htmlFor="">Email</label>
```
Comment (mishelshaji) [inline client/src/pages/student/Registration.js]: Give a value for the htmlFor property.

File: client/src/pages/student/Registration.js [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+                <input type="email" className="form-control" {...register("email")} />
+                {errors.email ? <span className="text-danger">{errors.email.message}</span> : null}
+              </div>
+              <div className="mb-3">
+                <label htmlFor="">Mobile</label>
```
Comment (mishelshaji) [inline client/src/pages/student/Registration.js]: Give a value for the htmlFor property.

File: client/src/pages/student/Registration.js [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+                <input type="text" className="form-control" {...register("mobile")} />
+                {errors.mobile ? <span className="text-danger">{errors.mobile.message}</span> : null}
+              </div>
+              <div className="mb-3">
+                <label htmlFor="">Password</label>
```
Comment (mishelshaji) [inline client/src/pages/student/Registration.js]: Give a value for the htmlFor property.

File: client/src/pages/student/Registration.js [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+                <input type="password" className="form-control" {...register("password")} />
+                {errors.password ? <span className="text-danger">{errors.password.message}</span> : null}
+              </div>
+              <div className="mb-3">
+                <label htmlFor="">Confirm Password</label>
```
Comment (mishelshaji) [inline client/src/pages/student/Registration.js]: Give a value for the htmlFor property.

**Review by mishelshaji — CHANGES_REQUESTED** [review]

**Review by mishelshaji — APPROVED** [review]

File: client/src/pages/instructor/Registration.js (hunk 2 above)
Comment (Nimishaer) [reply client/src/pages/instructor/Registration.js]: added single quotes to represent string value.

File: client/src/pages/instructor/Registration.js (hunk 4 above)
Comment (Nimishaer) [reply client/src/pages/instructor/Registration.js]: added single quotes to represent string value.

File: client/src/pages/instructor/Registration.js (hunk 3 above)
Comment (Nimishaer) [reply client/src/pages/instructor/Registration.js]: added single quotes to represent string value.

File: client/src/pages/instructor/Registration.js (hunk 5 above)
Comment (Nimishaer) [reply client/src/pages/instructor/Registration.js]: added single quotes to represent string value.

File: client/src/pages/instructor/Registration.js (hunk 6 above)
Comment (Nimishaer) [reply client/src/pages/instructor/Registration.js]: added single quotes to represent string value.

File: client/src/pages/instructor/Registration.js (hunk 7 above)
Comment (Nimishaer) [reply client/src/pages/instructor/Registration.js]: Added single quotes to represent string value.

File: client/src/pages/student/Registration.js (hunk 8 above)
Comment (Nimishaer) [reply client/src/pages/student/Registration.js]: Added single quotes to represent string value.

File: client/src/pages/student/Registration.js (hunk 9 above)
Comment (Nimishaer) [reply client/src/pages/student/Registration.js]: Added single quotes to represent string value.

File: client/src/pages/student/Registration.js (hunk 10 above)
Comment (Nimishaer) [reply client/src/pages/student/Registration.js]: Added single quotes to represent string value.

File: client/src/pages/student/Registration.js (hunk 11 above)
Comment (Nimishaer) [reply client/src/pages/student/Registration.js]: Added single quotes to represent string value.

File: client/src/pages/student/Registration.js (hunk 12 above)
Comment (Nimishaer) [reply client/src/pages/student/Registration.js]: Given value for htmlFor property.

File: client/src/pages/student/Registration.js (hunk 13 above)
Comment (Nimishaer) [reply client/src/pages/student/Registration.js]: Given value for htmlFor property

File: client/src/pages/student/Registration.js (hunk 14 above)
Comment (Nimishaer) [reply client/src/pages/student/Registration.js]: Given value for htmlFor property

File: client/src/pages/student/Registration.js (hunk 15 above)
Comment (Nimishaer) [reply client/src/pages/student/Registration.js]: Given value for htmlFor property

**Pull request merged**